Updated READMEs (#39)
AU-COBRA/ConCert: extraction/README.md

`theories` folder contains the implementation and correctness theorems.  The `examples` folder, as
the name suggests, contains examples of smart contracts and programs extracted using our development
and tests for our extensions to the certified erasure.

After building the project (running `make` from the project's root), the folders
`liquidity-extract/tests` and `elm-extract/tests` are populated with the extracted code. 

Compiling Liquidity code:
install the [Liquiditycompiler](https://www.liquidity-lang.org/doc/installation/index.html). Be
aware of some issues while installing Liquidity: https://github.com/OCamlPro/liquidity/issues/254.
Alternatively, the Liquidity code can be run using the online IDE: https://www.liquidity-lang.org/

Compiling Elm code:
install the [Elm compiler] (https://guide.elm-lang.org/install/elm.html).
Running Elm tests also requires `elm-explorations/test` package (see the required dependencies in
`elm-extract/elm.json`)

Some highlights from `theories`:

* `theories/ExAst.v` -- An extension of the MetaCoq's certified erasure EAst data-structures with additional information about erased types. 

> * `theories/Erasure.v` -- An extension of the MetaCoq's certified erasure with erasure for types. Also implements erasure for global environments with extra typing information for global definitions.

Reviewer: ```suggestion
* `theories/Erasure.v` -- An extension of the MetaCoq's certified erasure with erasure for types and erasing only required dependencies. Also implements erasure for global environments with extra typing information for global definitions.
```

> * `theories/ExTyping.v` -- Well-formedness of erased terms and erased global environments.

Reviewer: I think this file is not used anywhere as it is, we should be able to remove it (though I will probably use it for proving analysis correct in the future).

Author: I think it's imported in `OptimizeCorrectness.v`

Author: We still can keep it, but remove from highlights

Author: Damn, I forgot how to suggest changes :) @jakobbotsch could you delete the theories/ExTyping.v bit?

Reviewer: ```suggestion
```

> * `theories/ErasureCorrectness.v` -- Correctness lemmas for definition from `Erasure.v`.

Reviewer: ```suggestion
* `theories/ErasureCorrectness.v` -- Correctness lemmas for definitions from `Erasure.v`, proving that our erasure produces a well-formed erased environment.
```

* `theories/MetaCoqErasureCorrectnessStrong.v` -- A strengthened version of the correctness proof from MetaCoq (erasing dependencies only, not the full environment).
* `theories/Extraction.v` -- High-level interface to extraction. Provides different pipelines for doing extraction with different trusted computing bases.
* `theories/ExtractionCorrectness.v` -- Top-level correctness theorem relating the stages.
* `theories/Optimize.v` -- Optimisations (dead argument elimination, logical parameters elimination) on `λ□` terms.

> * `theories/OptimizeCorrectness.v` -- Optimsations correctness.

Reviewer: ```suggestion
* `theories/OptimizeCorrectness.v` -- Correctness of optimisation (dearging).
```

* `theories/CertifyingEta.v` -- A eta-expansion procedure which generated proofs of equality of between the result and the original.
* `theories/LPretty.v` -- Pretty-printer for Liquidity from `λ□`.
* `theories/Liquidity.v` -- A pretty printer that works directly on the deep embedding of `λsmart` language.
* `theories/LiquidityExtract.v` - A high-level interface to Liquidity extraction.

> * `theories/MidlangExtract.v` -- A high-level interface to Midlang extraction, incudes a pretty-printed and some examples.

Reviewer: ```suggestion
* `theories/MidlangExtract.v` -- A high-level interface to Midlang extraction, incudes a pretty-printer.
```
I deleted the examples.

Author: I'll bring them back :) I think we need examples for Midlang extraction. I tried with the counter - it works just fine.

Reviewer: I agree, but I think we should put them in examples, not in theories then.

Author: Ok, let's remove this stuff here, I'll put them in examples.
